src/app/blocks/db-block/db-block.component.ts

```diff
@@ -122,6 +122,43 @@ export class DbBlockComponent extends BaseBlockComponent {
         }.bind(this));
         return;
       }
+      case 'upsert': {
```

Comment:
Would it not be worth implementing the "Put" function for Dexie.js in the localDatabaseService? 
That handles all the update or replace logic too.

Reply:
Yes I started that and put it in a branch to investigate another time: https://github.com/kendraio/kendraio-app/tree/local-database-put-method